doc/architecture.md

```diff
@@ -0,0 +1,96 @@
+# Architecture
+
+The architecture of the system tests consist out of two layers.
```

Comment:
Wording seems to be weird `consists out of two`. Also, clarify that the layers are.